src/refract/RenderJSONVisitor.cc

```diff
@@ -9,265 +9,254 @@
+        {
+            const typename T::ValueType* val = GetValue<T>(element);
+
+            if (!val) {
+                return;
+            }
+
+            for (typename T::ValueType::const_iterator it = val->begin();
+                 it != val->end();
                  ++it) {
 
-                pArr.push(*it);
+                if (!(*it) || (*it)->empty()) {
+                    continue;
+                }
+
+                if ((*it)->element() == "ref") {
+                    IElement::MemberElementCollection::const_iterator found = (*it)->attributes.find("resolved");
+
+                    if (found == (*it)->attributes.end()) {
+                        continue;
+                    }
+
+                    const T* resolved = TypeQueryVisitor::as<T>((*found)->value.second);
+
+                    if (!resolved) {
+                        throw refract::LogicError("Mixin must refer to same type as parent");
+                    }
+
+                    FetchMembers(*resolved, members);
+                    continue;
+                }
+
+                RenderJSONVisitor renderer;
+                renderer.visit(*(*it));
+                IElement* e = renderer.getOwnership();
+
+                if (!e) {
+                    continue;
+                }
+
+                e->renderType(IElement::rCompact);
+                members.push_back(e);
+
             }
         }
+
     }
 
-    void RenderJSONVisitor::visit(const IElement& e)
-    {
+    RenderJSONVisitor::RenderJSONVisitor() : result(NULL), enumValue(NULL) {}
+
+    RenderJSONVisitor::~RenderJSONVisitor() { 
+        if (result) {
+            delete result;
+        }
+
+        if (enumValue) {
+            delete enumValue;
+        }
+    }
+
+    void RenderJSONVisitor::visit(const IElement& e) {
         e.content(*this);
     }
 
-    void RenderJSONVisitor::visit(const MemberElement& e)
-    {
+    void RenderJSONVisitor::visit(const MemberElement& e) {
+
         RenderJSONVisitor renderer;
 
         if (e.value.second) {
             if (IsTypeAttribute(e, "nullable") && e.value.second->empty()) {
-                renderer.result = sos::Null();
-            } else if (IsTypeAttribute(e, "optional") && e.value.second->empty()) {
+                renderer.result = new NullElement;
+            } 
+            else if (IsTypeAttribute(e, "optional") && e.value.second->empty()) {
                 return;
-            } else {
+            } 
+            else {
                 renderer.visit(*e.value.second);
+                if (!renderer.result) {
+                    return;
+                }
             }
         }
 
         if (StringElement* str = TypeQueryVisitor::as<StringElement>(e.value.first)) {
-            assign(str->value, renderer.get());
+            MemberElement *m = new MemberElement;
+            IElement* v= renderer.result ? renderer.getOwnership() : new StringElement;
+            m->set(str->value, v);
+            result = m;
```

Comment:
No, `result` is downcasted onto `IElement` so you will be not able to call `set()` method